### ISSUE TITLE
Improve expansion of date-type properties

### ISSUE THREAD

**Issue by BambanzaJuniorThe2nd** [alphamanuscript/surix-js-sdk#24]
For reasons unkown, the expansion of date-type properties returns incorrect results. For instance, if you had the following { 'Last Updated': 2018-01-06T22:00:00.000Z }, expandEntity would return ```{ 'Last Updated': { type: 'object', value: {} }```

As a fix, I propose you try setting the condition for date objects as follows `typeof date[key].getMonth === 'function'` instead of `date[key] instanceof Date`.

**Comment by grafitto** [comment]
@BambanzaJuniorThe2nd 
Do you mean:
```
{
    data: {
        'Last Updated': '2018-01-06T22:00:00.000Z'
    }
}
```
or
{
    data: {
        'Last Updated': new Date()
    }
}

**Comment by BambanzaJuniorThe2nd** [comment]
I mean ```{
    data: {
        'Last Updated': new Date()
    }
}```

**Comment by grafitto** [comment]
Nice. Looking into the issue

**Comment by habbes** [comment]
@grafitto @BambanzaJuniorThe2nd I picked into the code in the generate `dist/expand-entity.js` folder of the data-helpers library in node_modules and I found this:

<img width="424" alt="Screenshot 2019-06-15 at 23 49 48" src="https://user-images.githubusercontent.com/8460169/59556280-78893580-8fc8-11e9-99d0-43215d1aeb39.png">

So this does not match what's in the current code base, it does not account for arrays and dates in the `case 'object'` branch. Could it be that @grafitto did not update the library published on npm?

**Comment by grafitto** [comment]
I updated the package on npm and now the changes are reflected. The suggestions by @BambanzaJuniorThe2nd were implemented in #26 PR

Closing this issue
